chapter/related-work.md

```diff
@@ -12,27 +12,102 @@ Additionally, this part aims to recognize alternative approaches to the LSP, in
 
 #### Language Complexity
 
-#### LSP compliance
 
-#### Features
+### Representative LSP Projects
 
-#### File processing
+Since the number of implementations of the LSP is continuously growing, this thesis will present a selected set of notable projects.
 
-##### Incremental
+1. Three highly advanced and complete implementations that are the de-facto standard tooling for their respective language:
+   *rust-analyzer* [@rust-analyzer], *ocaml-lsp*/*merlin* [@ocaml-lsp,@merlin] and the *Haskell Language Server* [@hls]
+2. Two projects that provide compelling alternatives for existing specialized solutions:
+   *Metals* (for Scala) [@metals], *Java LSP* [@java-lsp]
+3. Language Servers for small languages in terms of complexity and userbase, highlighting one of the many use cases for the LSP.
+   *rnix-lsp* [@rnix-lsp], *frege-lsp*
 
-##### Full
 
-### Comparative Projects
 
-<!-- Rust Analyser -->
-<!-- Merlin -->
-<!-- rnix-lsp -->
-<!-- pylance? -->
-<!-- Scala & Java LSP (feature LSP extensions) -->
-<!-- haskell LSP? (talk to tim) -->
+#### LSP as standard tooling
+
+Today LSP based solutions serve as the go-to-method to implement language analysis tools.
+Especially emerging languages without existing integration with major IDEs make extensive use of the flexibility and reach provided by the LSP.
+
+Most languages profit greatly from the possibility to leverage existing tooling for the language.
+For instance the Haskell language server facilitates a plugin system that allows it to integrate with existing tooling projects [@hls-plugin-search].
+Plugins provide capabilities for linting [@hls-hlint-plugin], formatting [@hls-floskell-plugin,hls-ormolu-plugin] documentation [@hls-haddock-comments-plugin] and other code actions [@hls-tactics-plugin] across multiple compiler versions.
+While this requires HSL to be compiled with the same compiler version in use by the IDE, it also avoids large scale reimplementations of compiler features in an entirely different language.
```

Comment:
> Why this requires HSL to be compiled with the same compiler version in use by the IDE

I'm not sure this is relevant: it's a technical detail, and I imagine you would have the same issue even if you reimplement your own parser and analyzer: if the syntax changes, you can be out of sync between the compiler you are using for building and the parser used by the LSP. Maybe the restriction here is stricter, as here it wouldn't even start instead of just showing possible outdated diagnostics?